File: ObjC_Example/NativeVideo/NativeVideoAdLandscapeView.m
//
//  NativeVideoAdLandscapeView.m
//  ObjC_Example
//
//  Created by 馬場美沙都 on 2023/05/09.
//  Copyright © 2023 F@N Communications. All rights reserved.
//
#import "NativeVideoAdLandscapeView.h"

@implementation NativeVideoAdLandscapeView

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        [self loadXib];
    }
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)coder
{
    self = [super initWithCoder:coder];
    if (self) {
        [self loadXib];
    }
    return self;
}

- (void) loadXib
{
    UIView *view = [[[NSBundle mainBundle] loadNibNamed:@"NativeVideoAdLandscapeView" owner:self options:nil] firstObject];
    [self addSubview: view];
}

@end

File: Swift_Example/NativeVideo/View/NativeVideoAdLandscapeView.swift
//
//  NativeVideoAdLandscapeView.swift
//  Swift_Example
//
//  Created by 馬場美沙都 on 2023/04/18.
//  Copyright © 2023 F@N Communications. All rights reserved.
//

import UIKit

class NativeVideoAdLandscapeView: NativeVideoAdBaseView {
    
    override init(frame: CGRect){
        super.init(frame: frame)
        loadXib()
    }
    
    required init(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)!
        loadXib()
    }

    func loadXib(){
        let bundle = Bundle(for: type(of: self))
        let nib = UINib(nibName: "NativeVideoAdLandscapeView", bundle: bundle)
        let view = nib.instantiate(withOwner: self, options: nil).first as! UIView
        self.addSubview(view)
    }

}
